File: 2161. Partition Array According to Given Pivot/PartitionArrayAccordingToGivenPivot.java
public class PartitionArrayAccordingToGivenPivot {
    public int[] pivotArray(int[] nums, int pivot) {
        ArrayList<Integer> less = new ArrayList<>(),equal = new ArrayList<>(),more = new ArrayList<>();
        for (int i:nums){
            if (i<pivot){
                less.add(i);
            }
            else if(i>pivot){
                more.add(i);
            }
            else{
                equal.add(i);
            }
        }
        less.addAll(equal);
        less.addAll(more);
        return less.stream().mapToInt(i -> i).toArray();
    }
}

File: 2161. Partition Array According to Given Pivot/Partition Array According to Given Pivot.py
nums = [9,12,5,10,14,3,10]
pivot = 10
less = equal = large = 0
for i in nums:
    if i<pivot:
        less += 1
    elif i == pivot:
        equal += 1
large = (less+equal)
equal = less
less = 0
sol = [0]*len(nums)
for i in nums:
    if i<pivot:
        sol[less] = i
        less +=1
    elif i>pivot:
        sol[large] = i
        large +=1
    else:
        sol[equal] = i
        equal +=1
print(sol)